### ISSUE TITLE
``ColDefs.change_attribute()`` no longer works

### ISSUE THREAD

**Issue by weaverba137** [astropy/astropy#1539]
The ``change_attribute()`` method of ColDefs doesn't work as advertised.  A code snippet that demonstrates this is below.

```
import numpy as np
from astropy.io import fits

bits = 16
u0 = np.arange(17,dtype=np.uint16)
u = 2**u0 - np.uint16(1)
bzero = np.uint16(2**(bits-1))
col = fits.Column(name='u2',array=u,format='I',bzero=bzero)
print(col)
table = fits.new_table([col])
print(table.data.columns)
# This method call doesn't change bzero
table.data._coldefs.change_attrib('u2','bzero',0)
print(table.data.columns)
# This method does work.
table.data._coldefs[0].bzero = 0
print(table.data.columns)
```

**Comment by astrofrog** [comment]
@embray - should we remove the 1.0 milestone?

**Comment by astrofrog** [comment]
Removing the milestone since #3283 was not milestoned as 1.0

**Comment by embray** [comment]
Been thinking about/playing around with this, and I think I might actually try to fix this.  I still want to deprecate this method, but it should be possible to change attributes of FITS columns and have it actually reflect in the data.  This will be important even (if not especially) when we try to move FITS tables over to the `Table` interface.

That is, regardless of whether FITS table data is represented as a `FITS_rec` or a `Table`, it will be important to be able to change, say, the bzero attribute of a column and have that actually reflected.

**Comment by embray** [comment]
Fixed by #3283.
